sessions/2019/january/session_23.md

```diff
@@ -0,0 +1,149 @@
+
+Embrace this approach is expensive and requires big effort/time/money but have a very good [ROI][8] (get a lot in return).
+
+This begin with easy to read steps or test that are written usually in a folder called `features` inside the project's folder. A couple of examples can be seen [here][5] and [here][6]. In those links we can see that even if they are into different languages (Go and Java) libraries, the features definition is in the same language called Gherkin as was said.
+
+So, in the examples, some lists of features are defined. These are the high level steps legible for non technical people. For example:
+
+```text
+Feature: Configuring app version
+
+Background:
+    Given I set environment variable "API_KEY" to "a35a2a72bd230ac0aa0f52715bbdc6aa"
+    And I configure the bugsnag endpoint
+    And I set environment variable "APP_VERSION" to "3.1.2"
+    And I set environment variable "SERVER_PORT" to "4512"
+
+Scenario: A error report contains the configured app type when using a net http app
+    Given I set environment variable "AUTO_CAPTURE_SESSIONS" to "false"
+    When I start the service "nethttp"
+    And I wait for the app to open port "4512"
+    And I wait for 2 seconds
+    And I open the URL "http://localhost:4512/handled"
+    Then I wait to receive a request
+    And the request is a valid error report with api key "a35a2a72bd230ac0aa0f52715bbdc6aa"
+    And the event "app.version" equals "3.1.2"
+```
+
+A think to notice here, is that as the requirements change in the time, the test has to change also.
+
+Following to feature definition, in `features/steps` folder are the actual test implementations. There are the translation of the features to technical terms. So we can see some steps using Ruby language (see more [here][7] and [here][9]):
+
+```ruby
+require 'net/http'
+
+When(/^I wait for the app to open port "(.*)"$/) do |port|
+  wait_for_port(port)
+end
+
+Then(/^the request(?: (\d+))? is a valid error report with api key "(.*)"$/) do |request_index, api_key|
+  request_index ||= 0
+  steps %Q{
+    And the request #{request_index} is valid for the error reporting API
+    And the "bugsnag-api-key" header equals "#{api_key}" for request #{request_index}
+    And the payload field "apiKey" equals "#{api_key}" for request #{request_index}
+  }
+end
+```
+
+Finally, in `features/fixtures` folder are the running applications. Some Java applicacion is found [here][10] for example.
+
+#### Kanban feedback
+
+We fixed the kanban to this week. Some things were talked about:
+
+- When using kanban method consistency is more important, first we achieve a consistent behaviour and after we can increase values.
+
+- Was introduced the cocept of retrospective, we made a feedback session to see what happened during the week and what could be improved. Also was adviced to me that not always is needed to say or answer something to all that is said, that means one could just remain silent during feedback time.
```

Comment:
Simply put: You are encouraged to speak during a retro, but it is not required of you.